Run precheck make target in Travis CI

tklauser opened the pull request:
* move lock and rand checks to precheck target
* run build checks and complete build in travis

Please see individual commits for details.

Updates #11730

tklauser committed:
make: move lock and rand checks to precheck target

There's no reason to wait with running contrib/scripts/lock-check.sh and
contrib/scripts/rand-check.sh until after the build (i.e. the postcheck
target) as they are using only `grep` to check. Move them to the
precheck target so these errors are caught earlier.

Updates #11730

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

pchaigno commented on .travis/build.sh:
```diff
@@ -5,6 +5,7 @@ set -o errexit
 export CFLAGS="-Werror"
 export CGO_CFLAGS="-DCI_BUILD"
 
+make precheck
```
Couldn't we just run `make`?

tklauser replied:
Good point, changed in the latest revision. I was a bit worried that we might need additional build dependencies installed. But from a quick look everything seems to be there and otherwise we'll see from the run on this PR :smile:

pchaigno replied:
I figured since Travis CI is the only CI piece that tests aarch64, checking that it compiles there is probably worth the extra build time.

pchaigno replied:
Might be worth using `-j2` though. Travis CI usually runs on 2 cores and some part of the builds benefit from that.

tklauser committed:
travis: remove unused CGO_CFLAGS env var

The CI_BUILD define is no longer needed since commit 9886f71ad128
("pkg/bpf: remove #define depending on CI_BUILD")

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
travis: run build checks and complete build

This will allow various issues before actually running the tests in our
CI. Use `-j 2` since Travis CI usually runs on 2 cores and some parts of
the build benefit from that.

Updates #11730

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

christarazi reviewed (approved):
Nice!
